app/controllers/challenge_partnerships_controller.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+class ChallengePartnershipsController < ApplicationController
+  before_action :set_form, only: :create
+
+  def show
+    set_form
```

Comment:
why not put this `set_form` in `before_action` too?

Reply:
So that we can rescue in `show`

Reply:
I guess if we had redirects in `set_form` method, we could maybe simplify this significantly - no need to raise our custom errors if we redirect straightaway. Question is, will it work as expected?